### PR TITLE
Add reinhard color normalization support

### PR DESCRIPTION
**Pull request by andreped**
I have added support for reinhard color normalization, as an alternative to Macenko, as discussed in https://github.com/EIDOSLAB/torchstain/issues/24.

Reinhard itself was straight forward to implement, but I had a hard time getting the RGB/LAB color converters to work as intended. After looking at what OpenCV, Scikit-Image, and other projects were doing, I managed to implement numpy-equivalents. In the end, from qualitative assessment, I get the same results as another commonly used library, [StainTools](https://github.com/Peter554/StainTools).

Have not had the time to benchmark it yet, hence, `experimental`, but at least it runs and serves as an alternative method. 

**UPDATE: Now both Numpy and TensorFlow backends are supported!**

## What's changed:
- Added `ReinhardNormalizer` - currently supports both numpy and tensorflow backends.
- Added `rgb2lab` and `lab2rgb` color conversion implementations.
- Added convenience functions relevant to Reinhard in `stats` and `split`.
- Updated README regarding `experimental Reinhard support` with reference to original article.
- Added RGB->LAB->RGB unit test (currently fails - focused on getting it to work with Reinhard and not in the general case).
- Moved `result.png`to `data/`directory and updated corresponding path in README.
- Minor README refactoring.

### CODE REVIEW

**Comment by andreped** [comment]
Note that I also added TF support now, as I need it for one of my studied.

Will not have time for PyTorch any time soon, but at least both Numpy and TF backends work as intended.

**Comment by carloalbertobarbano** [comment]
Thanks for the contribution! Merging it now
